[MIG][12.0] l10n_br_tef

mileo opened the pull request:
Esse modulo integra com o TEF da http://vspague.com.br/

O correto é dividir ele em dois, um modulo só com os fluxos e partes básicas e outros para cada integração, por exemplo:

- l10n_br_tef_vspague
- l10n_br_tef_sitef
- l10n_br_tef_etc

Extraido de https://github.com/kmee/l10n-brazil/pull/611 a pedido do Rafael da https://github.com/cybertrongw

Temos previsão de continuar a migração do mesmo a e integração com o SITEF: https://www.softwareexpress.com.br/pt/solucoes/SiTef/

hugouchoasborges committed:
[ADD] Módulo criado

hugouchoasborges committed:
[WIP] Envios de mensagens pelo javascript

hugouchoasborges committed:
[IMP] Criada fila de transações tornando possível reenviar uma transação

hugouchoasborges committed:
[WIP] Criados estados para fluxo de compra

atillaSilva committed:
[ADD] Tipo de cartao, campo selected

atillaSilva committed:
[ADD] todos os campos do tef

Commit:
[RMV] Removendo campos desnecessários

Commit:
[ADD] Primeira versão do TEF

hugouchoasborges committed:
[ADD] Funcionamento do TEF sem o PinPad

Commit:
[FIX] erro de escolha de campo com varios meios de pagamento

Commit:
[IMP] tratando cancelamento de operacao

Commit:
[FIX] 'Credit' function Working with\without pinpad

Commit:
[ADD] Função Débito

Commit:
[ADD] Tratamento de excessões

atillaSilva committed:
[ADD] Auto-connect ao PinPad

atillaSilva committed:
[ADD] mensagens para pagamentos com o tef

Commit:
[ADD] Mensagens padrões na tela do POS

Commit:
[ADD] Métodos de pagamento padrões - Crédito\Débito

Commit:
[FIX][IMP] Tratamento de exceções e Atualizaçao dos servidores de teste

hugouchoasborges committed:
[FIX] Selecionando valor do pagamento

hugouchoasborges committed:
[WIP] Removendo dados de demonstração

hugouchoasborges committed:
[ADD] Parcelamento

hugouchoasborges committed:
[ADD] Tradução

hugouchoasborges committed:
[FIX] Necessidade de clicar em iniciar transação 2 vezes

hugouchoasborges committed:
[ADD][WIP] Tela de cancelamento de compras

Commit:
[IMP] README atualizado

Commit:
[FIX] Verifica se V$Pague está online antes de iniciar transação

Commit:
[ADD] Inserção manual de cartão\validade\cod.segurança no modo Debug

Commit:
[FIX] Problema na conexão ao iniciar POS

Commit:
[IMP] Atualização no servidor da VBI

Commit:
[IMP] Tradução atualizada

Commit:
[WIP] Cancelamento de Compras ambiente homologação - Transação não localizada

Commit:
[FIX] Cor do ícone do TEF e Bugs

Commit:
[ADD] Criado grupo "Configurações TEF" nas configurações do POS

Commit:
[ADD] Campos de configuração do servidor de homologação

Commit:
[IMP] Tradução Atualizada

Commit:
[IMP] Produção funcionando
Apenas MASTERCARD - Crédito

Commit:
[ADD] Adicionada visão Popup para confirmação de cancelamento

Commit:
[IMP][FIX] Resetando valores de variáveis relacionadas ao cancelamento após abortar operação

Commit:
[ADD] Chamando visão de confirmação de cancelamento de compras

Commit:
[ADD] Botão "não" funcionando no cancelamento

Commit:
[IMP] Popup de confirmação de cancelamento mostrando informações da compra

Commit:
[IMP] Unbind onclick

Commit:
[IMP] RenderElement na tela de confirmação de cancelamento funcionando

Commit:
[ADD] Cancelamento funcionando (mastercard)

mileo committed:
[MIG] l10n_br_tef: Migration to 12.0

mileo committed:
[REF] l10n_br_tef: Black python code

mileo committed:
[REF] pre-commit l10n_br_tef

mileo committed:
[REF] Adapt files to OCA pattern

mileo committed:
[MIG] Module data and demo

mileo committed:
[MIG] Make main tef processes work

luismalta committed:
[FIX] TEF Status Popups

luismalta committed:
[FIX] TEF Debug card selector

luismalta committed:
[FIX] Credit payment flow

luismalta committed:
[FIX] Debit payment flow

luismalta committed:
[ADD] Validation of lines already paid for by TEF

luismalta committed:
[FIX] Transaction abort flow

marcelsavegnago commented:
@luismalta se possível roda um pre-commit para permitir que os demais testes sejam executados. Vlww

luismalta committed:
[REF] Remove l10n_br_tef.js

Cancel popups and order list screen will be implemented differently for version 12.0

luismalta committed:
[REF] l10n_br_tef: Improves screens.js code

luismalta committed:
[REF] l10n_br_tef: improve models.js code

luismalta committed:
[REF] l10n_br_tef: improve widgets.js code

luismalta committed:
[REF] l10n_br_tef: pos_assets.xml

luismalta committed:
[REF] l10n_br_tef: Remove global variable

luismalta committed:
[FIX] Manual card data input flow

luismalta committed:
[REF] l10n_br_tef: Remove global variable

luismalta committed:
[ADD] l10n_br_tef: get selected payment line function

luismalta committed:
[ADD] l10n_br_tef: pos_order_mgmt dependency

luismalta committed:
[ADD] ROADMAP.rst

luismalta committed:
[IMP] l10n_br_tef_destaxa: pre-commit
